stixpy/product/scratch.py

```diff
@@ -14,12 +37,24 @@
 times = Time("2024-01-01T00:00:00") + np.arange(pd_shape[0]) * 2 * u.s
 energies = (1 + 2 * np.arange(pd_shape[-1])) * u.keV
 
-# Create a gwcs
-energy_coord = QuantityTableCoordinate(energies, names="energy", physical_types="em.energy")
+# Create table coords
 time_coord = TimeTableCoordinate(times, names="time", physical_types="time")
+energy_coord = QuantityTableCoordinate(energies, names="energy", physical_types="em.energy")
+
+# fake coords for detector and pixel
+detector_coord = QuantityTableCoordinate(
+    np.arange(pd_shape[1]) * u.dimensionless_unscaled, names="detector", physical_types="custom:detector"
+)
+pixel_coord = QuantityTableCoordinate(
+    np.arange(pd_shape[2]) * u.dimensionless_unscaled, names="pixel", physical_types="custom:pixel"
+)
+
+combine_coord = energy_coord & pixel_coord & detector_coord & time_coord
 
-combine_coord = time_coord & energy_coord
 
+# I'm not sure what should live in the meta vs extra_coords
```

Comment:
In ndcube-land, a coordinate is a strictly monotonic sampling of a coordinate space, i.e., the values must be numerical, ordered and strictly increasing or decreasing.  If your property doesn't satisfy this, then it's axis-aligned metadata.  Once [ndcube #455](https://github.com/sunpy/ndcube/pull/455) is merged and ndcube 2.3 released, `NDCube` will support axis-aligned metadata.  This will be the appropriate place for detector labels, as they will be sliced when the cube is sliced.  And we can add functionality to a `class STIXMeta(ndcube.Meta)` so that detector labels are properly combined when the data is rebinned.